filecheck/checker.ts

```diff
@@ -185,34 +241,86 @@ export async function checkFile(
         );
         fse.copyFileSync(compressed.destinationPath, filePath);
       } else {
-        const msg = `${filePath} is ${formatSize(
-          sizeBefore
-        )} can be compressed to ${formatSize(
-          sizeAfter
-        )} (${reductionPercentage.toFixed(0)}%)`;
-        console.warn(msg);
-        console.log(
-          "Consider running again with '--save-compression' and run again " +
-            "to automatically save the newly compressed file."
-        );
-        const relPath =
-          process.env.CI === "true"
-            ? path.relative(process.cwd(), filePath)
-            : filePath;
-        const cliCommand = `yarn filecheck "${relPath}" --save-compression`;
-        console.log(`HINT! Type the following command:\n\n\t${cliCommand}\n`);
-        throw new Error(
-          `${filePath} can be compressed by ~${reductionPercentage.toFixed(0)}%`
+        throw new FixableError(
+          `${filePath} is ${formatSize(
+            sizeBefore
+          )} and can be compressed to ${formatSize(
+            sizeAfter
+          )} (${reductionPercentage.toFixed(0)}%)`,
+          `yarn filecheck --save-compression '${getRelativePath(filePath)}'`
         );
       }
     }
-  } catch (error) {
+  } finally {
     fse.removeSync(tempdir);
-    console.error(error);
-    throw error;
   }
 }
 
-export async function runChecker(files: string[], options: CheckerOptions) {
-  return Promise.all(files.map((f) => checkFile(f, options)));
+function canCheckFile(filePath: string) {
+  return (
+    /\/files\//.test(filePath) &&
+    !/\/node_modules\//.test(filePath) &&
+    !/\.(DS_Store|html|json|md|txt|yml)$/i.test(filePath)
+  );
+}
+
+async function resolveDirectory(file: string): Promise<string[]> {
+  const stats = await fs.lstat(file);
+  if (stats.isDirectory()) {
+    const api = new fdir()
+      .withErrors()
+      .withFullPaths()
+      .filter((filePath) => canCheckFile(filePath))
+      .crawl(file);
+    return api.withPromise() as Promise<PathsOutput>;
+  } else if (stats.isFile() && canCheckFile(file)) {
+    return [file];
+  } else {
+    return [];
+  }
+}
+
+export async function runChecker(
+  filesAndDirectories: string[],
+  options: CheckerOptions
+) {
+  const errors = [];
+
+  const files = (
+    await Promise.all(filesAndDirectories.map(resolveDirectory))
+  ).flat();
+
+  const progressBar = new cliProgress.SingleBar(
+    { etaBuffer: 100 },
+    cliProgress.Presets.shades_grey
```

Comment:
I just used this, because we already use it in the build CLI:

https://github.com/mdn/yari/blob/fcce25f4d18c28c885456904d715d6a8f72b21d3/build/cli.ts#L122-L125

But I'll use the default here.